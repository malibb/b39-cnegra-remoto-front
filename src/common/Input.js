import React from 'react';

function Input({label,type,placeholder,change,value,name,required}){
    return(
        <div className="control-group">
            <div className="form-group floating-label-form-group controls">
              <label>{label}</label>
              <input 
              type={type}
              name={name}
              className="form-control" 
              placeholder={placeholder} 
              value={value}
              required={required}
              onChange={change}
              />
              <p className="help-block text-danger"></p>
            </div>
        </div>
    );
};

export default Input;