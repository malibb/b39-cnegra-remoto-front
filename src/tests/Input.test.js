import React from 'react';
import { shallow } from 'enzyme';
import Input from '../common/Input';

//shallow permitir props, state

describe('<Input/>', () => { // Test suite
    it('Simple Rendering', () =>{
        const component = shallow(<Input/>);
        expect(component).toMatchSnapshot();
    });

    it('test props', ()=> {
        const component = shallow(<Input label="Nombre:"/>);
        expect(component.find('.title-label').text()).toBe("Nombre:");
    })
});