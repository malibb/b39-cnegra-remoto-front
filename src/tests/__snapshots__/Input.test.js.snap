// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Simple Rendering 1`] = `
<div
  className="control-group"
>
  <div
    className="form-group floating-label-form-group controls"
  >
    <label
      className="title-label"
    />
    <input
      className="form-control"
    />
    <p
      className="help-block text-danger"
    />
  </div>
</div>
`;
